docs/authentication.md

```diff
@@ -32,35 +34,48 @@ The Helios client will now need to present a client certificate to authenticate.
 This can be either a pre-issued X.509 certificate file and associated private
 key, or a certificate generated based on your SSH key.
 
-### Existing X.509 certificate
-
-This is actually not yet implemented.
+First we'll explain briefly how the ssh-agent signed certificates work, and then go through how to supply credentials for the `helios` CLI and the Helios API respectively.
 
 ### On-the-fly certificates with ssh-agent
 
 If you have an OpenSSH key and use [ssh-agent](http://linux.die.net/man/1/ssh-agent)
-(you probably do if you use `ssh`), Helios can generate an X509 certificate for
+(you probably do if you use `ssh`), Helios can generate an X.509 certificate for
 you on the fly. Here's how it works:
 
 1. The Helios client checks for a running SSH agent, and if one is found, asks
    for a list of your public SSH keys. Private keys are never exposed by
    ssh-agent.
 
-2. The Helios client generates a self-signed X.509 certificate containing your
-   SSH public key and username, which it sends to the HTTPS server.
+2. The Helios client generates a temporary X.509 certificate containing your username.
```

Comment:
Yeah, but it's tricky given that it depends on whether you're using the CLI or the API. For the API you have to set the user explicitly, and for the CLI it comes from the `--username` arg which defaults to `System.getProperty("user.name")`.